src/util/color/predefinedcolorset.h

```diff
@@ -0,0 +1,143 @@
+#ifndef COLOR_H
+#define COLOR_H
+
+#include <QColor>
+#include <QObject>
+
+#include "util/color/predefinedcolor.h"
+#include "util/color/predefinedcolormap.h"
+
+// This class defines a set of predefined colors and provides some handy functions to work with them.
+// A single global instance is create in the Color namespace.
+// This class is thread-safe because all its methods and public properties are const.
+class PredefinedColorSet final {
+  public:
+    const PredefinedColorPointer noColor = std::make_shared<PredefinedColor>(
+        QColor(),
+        QLatin1String("No Color"),
+        QObject::tr("No Color"),
+        0
+    );
+    const PredefinedColorPointer red = std::make_shared<PredefinedColor>(
+        QColor("#E6194B"),
+        QLatin1String("Red"),
+        QObject::tr("Red"),
+        1
+    );
+    const PredefinedColorPointer green = std::make_shared<PredefinedColor>(
+        QColor("#3CB44B"),
+        QLatin1String("Green"),
+        QObject::tr("Green"),
+        2
+    );
+    const PredefinedColorPointer yellow = std::make_shared<PredefinedColor>(
+        QColor("#FFE119"),
+        QLatin1String("Yellow"),
+        QObject::tr("Yellow"),
+        3
+    );
+    const PredefinedColorPointer blue = std::make_shared<PredefinedColor>(
+        QColor("#4363D8"),
+        QLatin1String("Blue"),
+        QObject::tr("Blue"),
+        4
+    );
+    const PredefinedColorPointer cyan = std::make_shared<PredefinedColor>(
+        QColor("#42D4F4"),
+        QLatin1String("Cyan"),
+        QObject::tr("Cyan"),
+        5
+    );
+    const PredefinedColorPointer magenta = std::make_shared<PredefinedColor>(
+        QColor("#F032E6"),
+        QLatin1String("Magenta"),
+        QObject::tr("Magenta"),
+        6
+    );
+    const PredefinedColorPointer pink = std::make_shared<PredefinedColor>(
+        QColor("#FABEBE"),
+        QLatin1String("Pink"),
+        QObject::tr("Pink"),
+        7
+    );
+    const PredefinedColorPointer teal = std::make_shared<PredefinedColor>(
+        QColor("#469990"),
+        QLatin1String("Teal"),
+        QObject::tr("Teal"),
+        8
+    );
+    const PredefinedColorPointer grey = std::make_shared<PredefinedColor>(
+        QColor("#A9A9A9"),
+        QLatin1String("Grey"),
+        QObject::tr("Grey"),
+        9
+    );
+
+    // The list of the predefined colors.
+    const QList<PredefinedColorPointer> allColors {
```

Comment:
how about hide this as implementation detail and just expose the used functions like at() and count()